app/src/test/java/com/asfoundation/wallet/promotions/PromotionsRepositoryTest.kt

```diff
@@ -0,0 +1,284 @@
+
+    Assert.assertEquals(forecastBonus, testForecastBonus)
+  }
+
+  @Test
+  fun getForecastBonusInactiveTest() {
+    Mockito.`when`(
+        api.getForecastBonus(TEST_WALLET_ADDRESS, "packageName", BigDecimal.TEN, "APPC"))
+        .thenReturn(Single.just(ForecastBonusResponse(BigDecimal.ONE, TEST_GAMIFICATION_LEVEL,
+            ForecastBonusResponse.Status.INACTIVE)))
+    val testForecastBonus = ForecastBonus(ForecastBonus.Status.INACTIVE, BigDecimal.ZERO, "")
+    val forecastBonus =
+        promotionsRepository.getForecastBonus(TEST_WALLET_ADDRESS, "packageName", BigDecimal.TEN)
+            .blockingGet()
+
+    Assert.assertEquals(forecastBonus, testForecastBonus)
+  }
+
+  @Test
+  fun getGamificationStatsTest() {
+    mockUserStatsCalls()
+
+    Mockito.`when`(local.setGamificationLevel(TEST_GAMIFICATION_LEVEL))
+        .thenReturn(Completable.complete())
+
+    val testGamificationStats =
+        GamificationStats(GamificationStats.Status.OK, TEST_GAMIFICATION_LEVEL,
+            gamificationResponse.nextLevelAmount, gamificationResponse.bonus,
+            gamificationResponse.totalSpend, gamificationResponse.totalEarned, true)
+    val gamificationStats = promotionsRepository.getGamificationStats(TEST_WALLET_ADDRESS)
+        .blockingGet()
+
+    Mockito.verify(local)
+        .setGamificationLevel(TEST_GAMIFICATION_LEVEL)
+    Assert.assertEquals(testGamificationStats, gamificationStats)
+    assertUserStatsCalls()
+  }
+
+  @Test
+  fun getLevelsTest() {
+    val response = LevelsResponse(listOf(Level(BigDecimal.TEN, 20.0, TEST_GAMIFICATION_LEVEL)),
+        LevelsResponse.Status.ACTIVE, null)
+    Mockito.`when`(api.getLevels(TEST_WALLET_ADDRESS))
+        .thenReturn(Single.just(response))
+    Mockito.`when`(local.deleteLevels())
+        .thenReturn(Completable.complete())
+    Mockito.`when`(local.insertLevels(response))
+        .thenReturn(Completable.complete())
+
+    val testLevels = Levels(Levels.Status.OK,
+        listOf(Levels.Level(BigDecimal.TEN, 20.0, TEST_GAMIFICATION_LEVEL)), true, null)
+    val levels = promotionsRepository.getLevels(TEST_WALLET_ADDRESS)
+        .blockingGet()
+
+    Assert.assertEquals(testLevels, levels)
+    Mockito.verify(local)
+        .deleteLevels()
+    Mockito.verify(local)
+        .insertLevels(response)
+  }
+
+  @Test
+  fun getLevelsApiErrorTest() {
+    val response = LevelsResponse(listOf(Level(BigDecimal.TEN, 20.0, TEST_GAMIFICATION_LEVEL)),
+        LevelsResponse.Status.ACTIVE, null)
+    Mockito.`when`(api.getLevels(TEST_WALLET_ADDRESS))
+        .thenReturn(Single.error(Throwable("Any")))
+    Mockito.`when`(local.getLevels())
+        .thenReturn(Single.just(response))
+
+    val testLevels = Levels(Levels.Status.OK,
+        listOf(Levels.Level(BigDecimal.TEN, 20.0, TEST_GAMIFICATION_LEVEL)), true, null)
+    val levels = promotionsRepository.getLevels(TEST_WALLET_ADDRESS)
+        .blockingGet()
+
+    Assert.assertEquals(testLevels, levels)
```

Comment:
In this case we can also test that there were no interactions to deleteLevels and insertLevels